aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -224,13 +226,13 @@ NestedTensorImpl::NestedTensorImpl(
     const at::Tensor& base_tensor,
     at::Tensor nested_size_tensor,
     at::Tensor nested_stride_tensor,
-    std::vector<int64_t>&& offsets)
+    at::Tensor offsets_tensor)
```

Comment:
Other random comment: I still want a cpp expert to infer if we should taking in rvalue references for all pieces of metadata. Alban said that moving the temps will avoid refcount bumps and since we are (for now) assuming the callers of the constructor will create new unique pieces of metadata then we should be using rvalues.

But still clang tidy will say: `Moving a temporary object prevents copy elision (fix available)` So idk. PREMATURE OPTIMIZATION FOR THE WIN!

Reply:
Did we determine this is fine or does something need to change?

Reply:
This is goood